xova/apps/xova/app.py

```diff
@@ -40,6 +22,24 @@ def warn_with_traceback(message, category, filename, lineno, file=None, line=Non
 from xova.apps.xova.subtables import copy_subtables
 from xova.apps.xova.fixvis import fixms
 
+
+def warn_with_traceback(message, category, filename, lineno, file=None, line=None):
+    log = file if hasattr(file, 'write') else sys.stderr
+    traceback.print_stack(file=log)
+    log.write(warnings.formatwarning(message, category, filename, lineno, line))
+
+
+warnings.showwarning = warn_with_traceback
+
+
+try:
+    import bokeh  # noqa
+except ImportError:
+    can_profile = False
+else:
+    can_profile = True
+
+
```

Comment:
just move this stuff. Flake8 moans about imports not at the top of the file. Could possibly just tell it shut up on that one

Reply:
Also w.r.t. to pre-commit hooks, we just use the black autoformatter to avoid having to chase down flake8 errors all the time. But this is for a future PR and release IMO.

Reply:
we could -- I'm just a bit skeptical about running black automatically without supervision. It has a tendency of really messing up code style in places.